tests/ops/op_math/test_prod.py

```diff
@@ -891,6 +891,15 @@ def test_diagonalizing_gates(self):
         ),
     )
 
+    def test_pauli_rep_order(self):
```

Comment:
I am a bit puzzled because these tests now fail on the branch of [this PR](https://github.com/PennyLaneAI/pennylane/pull/5132) even though they should be unrelated and I pulled the latest `master` branch 🤔 
https://github.com/PennyLaneAI/pennylane/pull/5132